web/src/settings_notifications.ts

```diff
@@ -216,88 +237,101 @@ export function set_up(settings_panel) {
 
     // Common handler for sending requests to the server when an input
     // element is changed.
-    $container.find(".notification-settings-form").on("change", "input, select", function (e) {
-        e.preventDefault();
-        e.stopPropagation();
-        const $input_elem = $(e.currentTarget);
-        if ($input_elem.parents("#stream-specific-notify-table").length) {
-            stream_notification_setting_changed(e.currentTarget);
-            return;
-        }
-        const setting_name = $input_elem.attr("name");
-
-        if ($input_elem.attr("data-setting-widget-type") === "time-limit") {
-            // For time-limit settings we should always pass the select element
-            // to get_input_element_value and not the custom input element.
-            const select_elem = $input_elem
-                .closest(".time-limit-setting")
-                .find("select.settings_select")[0];
-            const setting_value = settings_components.get_input_element_value(select_elem);
-
-            // Currently only notification batching setting is the time-limit
-            // settings on this page.
-            if (
-                setting_name === "email_notifications_batching_period_seconds" &&
-                $input_elem.val() === "custom_period"
-            ) {
-                set_notification_batching_ui(
-                    $container,
-                    settings_object.email_notifications_batching_period_seconds,
-                    true,
+    $container
+        .find(".notification-settings-form")
+        .on("change", "input, select", function (this: HTMLElement, e) {
```

Comment:
We can prevent Prettier from indenting this whole function an extra level by extracting a variable:

```js
    const $notification_settings_form = $container.find(".notification-settings-form");
    $notification_settings_form.on("change", "input, select", function (this: HTMLElement, e) {
```